Public/Src/Cache/ContentStore/Test/Utils/NagleQueueTests.cs

```diff
@@ -12,6 +12,57 @@ namespace ContentStoreTest.Utils
 {
     public class NagleQueueTests
     {
+        [Fact]
+        public void ResumeShouldTriggerBatchOnTime()
+        {
+            bool processBatchIsCalled = false; ;
+            var queue = NagleQueue<int>.Create(
+                processBatch: data =>
+                              {
+                                  processBatchIsCalled = true;
+                                  return Task.FromResult(42);
+                              },
+                maxDegreeOfParallelism: 1,
+                interval: TimeSpan.FromMilliseconds(1),
+                batchSize: 10);
+
+            var suspender = queue.Suspend();
+            Thread.Sleep(1000);
+            queue.Enqueue(42);
+            suspender.Dispose(); // This should resume the queue and restart the timer
+
+            Thread.Sleep(1000); // Definitely longer than the configured interval provided to NagleQueue
+
+            // It means that the queue should call the callback and we can rely on that.
+
+            Assert.True(processBatchIsCalled);
+        }
+
+        [Fact]
+        public void EnqueingItemsInfrequentlyShouldAlwaysTriggerCallbackOnTime()
+        {
+            int processBatchIsCalled = 0;
+            var queue = NagleQueue<int>.Create(
+                processBatch: data =>
+                              {
+                                  processBatchIsCalled++;
+                                  return Task.FromResult(42);
+                              },
+                maxDegreeOfParallelism: 1,
+                interval: TimeSpan.FromMilliseconds(10),
+                batchSize: 10);
+
+            queue.Enqueue(42);
+            
+            Thread.Sleep(100);

```

Comment:
These `Sleep` calls will lead to flakiness, right? These tests tend to run at 100% CPU, so there's a chance the queue might not complete its operation before the assertion is called. Does it make sense to decrease the queue's interval even further to 1ms?